// common/dist/index.js
import * as z from 'zod';
export const InputModel = z.object({
    username: z.string(),
    password: z.string().min(8),
    type: z.enum(["Admin", "User"])
});
export const userMetaData = z.object({ avatarId: z.string() });
export const spaceData = z.object({ name: z.string(),
    dimensions: z.string(),
    mapId: z.string()
});
export const spaceElements = z.object({
    elementId: z.string(),
    spaceId: z.string(),
    x: z.number(),
    y: z.number()
});
export const mapCreator = z.object({
    imageUrl: z.string(),
    width: z.number(),
    height: z.number(),
    static: z.boolean()
});
export const updateImage = z.object({
    imageUrl: z.string()
});
export const avatarInputs = z.object({
    imageUrl: z.string(),
    name: z.string()
});
export const adminMap = z.object({
    thumbnail: z.string(),
    dimensions: z.string(),
    name: z.string(),
    defaultElements: z.array(z.object({
        elementId: z.string(),
        x: z.number(),
        y: z.number()
    }))
});
